best-practices/native/react-native/expo/upgrade/README.md

```diff
@@ -0,0 +1,92 @@
+
+## Expo upgrade process
+
+On the [official walkthrough for upgrading the Expo SDK](https://docs.expo.io/versions/latest/workflow/upgrading-expo-sdk-walkthrough/) they **strongly recommend** that you upgrade your SDK incrementally, instead of going straight from 32 to 36.
+
+They also list breaking changes for each step which you should definitely study. The big one is the import structure change from the first step.
+
+You will also want to **make sure your Expo cli is current**. You can `npm uninstall -g expo-cli` and `npm install -g expo-cli` just to be certain. The first stage of the upgrade (32 to 33) did not work correctly when I used an older cli.
+
+## 32 to 33
+
+The instructions on the site worked perfectly here, update the following dependencies in the `app.json` file:
+
+```
+{
+  "react-native": "https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz",
+  "expo": "^33.0.0",
+  "react": "16.8.3"
+}
+```
+
+Delete the node_modules directory and perform another npm install. If you have the Expo app installed on the simulator, you will need to delete it, and let the builder re-install the Expo app when you attempt to run the app. Failing to do this will result in a React Native version mismatch. There is one other time where this is required.
+
+Be sure to read the notes on the modular format for Expo imports. You will likely need to run `expo install expo-*package-name*` for each one. A trick I used was to search your project for `from 'expo';`. There are a couple of packages where this doesn't apply (Notifications is one of them), but for the most part you will need to make these changes. If you run the app after this upgrade without adjusting to the modular format, you may get a nice warning in the builder telling you about the offending package.
+
+With this step complete, you should (as of Feb. 2020) be able to have a working dev environment.
+
+## 33 to 34
+
+If you attempt to upgrade the SDK without committing the changes from the first update, you will get a warning about making changes on a dirty branch. It is very much recommended that you make a new branch for each stage of the upgrade. You can always delete the unneeded branches when you are done.
+
+The command listed on the walkthrough should now work, `expo upgrade 34.0.0`. You will get questions about upgrading your simulators, I answered no to both.
+
+At this point, things start to get a little dicey. The first error you will likely encounter involves custom fonts; you will get an error saying that the font is not properly loaded through `Font.loadAsync`, even if you are doing so properly. To resolve this, you must take the following steps:
+
+ - In your `package.json` make the following changes
+    - `expo` should be `34.0.3`
```

Comment:
I think in this case it's not a big a deal since the expo version will immediately be changed with the next command, but since this is caused by a caching error, being explicit is still a good idea. Fixed it.

Reply:
I actually meant the opposite of that (although you said it's not a big deal). Per the [npm documentation](https://docs.npmjs.com/misc/semver#tilde-ranges-123-12-1), if there were a `1.2`, a bunch of point releases (`1.2.1`, `1.2.2`, etc), and a `1.3`, if you specify (`~1.2`), npm would be happy to update you to `1.2.2` silently. Often we want this because updating the version numbers for tiny changes often isn't worth it.

The reason I recommended having a note telling the developer to make sure there was no `~` is simply because this upgrade process is very finicky. It doesn't sound like it's critical here.